ocaml/typing/parmatch.ml

```diff
@@ -2209,10 +2212,10 @@ let inactive ~partial pat =
    on exhaustive matches only.
 *)
 
-let check_partial pred loc casel =
+let check_partial ?(should_be_exhaustive = true) pred loc casel =
```

Comment:
I'm strongly averse to introducing new uses of optional arguments. (I sent you a link.) I'd much rather be confident that all callsites consider what behavior they want. I see existing uses of optional arguments in `typecore` as an anti-pattern. If it's easy, can we make this a required argument?

Separately, let's use a different name here. I kind of like `~warn_if` that can be either `Partial` or `Total`. Generally, it seems like the convention in typecore and parmatch is to refer to the concept as "partial vs. total" instead of "exhaustive vs. non-exhaustive" (though I'll grant that "exhaustive" is what's exposed to the end-user via the text of the warning).